apps/aecore/src/aec_conductor.erl

```diff
@@ -854,14 +859,27 @@ handle_add_block(Header, CheckFun, Block, State, Origin) ->
                             maybe_publish_block(Origin, Block),
                             case preempt_if_new_top(State, Origin) of
                                 no_change ->
-                                    {ok, State};
+                                    {ok, State#state{pof = no_fraud}};
```

Comment:
Don't we loose the PoF here? In the case we are not yet the leader but becomes one a couple of micro_blocks later? Not 100% sure though...

Reply:
Well, solid tests are a must ... 😛